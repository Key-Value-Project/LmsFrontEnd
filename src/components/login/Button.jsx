// eslint-disable-next-line react/prop-types
const Button = ({ text, error }) => {
	const style = {
		background: "red",
	};

	return <button style={error ? style : { cursor: "pointer" }}>{text}</button>;
};

export default Button;
