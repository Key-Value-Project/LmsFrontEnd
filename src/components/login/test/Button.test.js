/* eslint-disable no-unused-vars */
/* eslint-disable no-undef */
import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import Button from '../Button';

describe('Check Login Button', () => {
  test('Rendered', () => {
    const { getByTestId } = render(<Button />);
    const button = getByTestId('test-button-login');
    expect(button).toBeTruthy();
  });

  test('Text On Button Displayed', () => {
    const { getByTestId } = render(<Button text="OK" />);
    const button = getByTestId('test-button-login');
    expect(button.textContent).toBe('OK');
  });

  // test("Button onClick Trigger", () => {
  //     const onClick = jest.fn();
  // 	const { getByTestId } = render(<Button onClick={onClick} />);
  // 	const button = getByTestId("test-button-login");
  // 	fireEvent.click(button);
  //     expect(onClick).toHaveBeenCalledTimes(1);
  // });

  test('CSS inline loaded', () => {
    const { getByTestId } = render(<Button error={false} />);
    const button = getByTestId('test-button-login');
    expect(button).toHaveStyle('cursor: pointer');
  });

  test('Snapshot Matching', () => {
    const { asFragment } = render(<Button text="OK" />);
    expect(asFragment()).toMatchSnapshot();
  });
});
