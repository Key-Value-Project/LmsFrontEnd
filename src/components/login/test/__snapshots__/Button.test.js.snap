// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Check Login Button Snapshot Matching 1`] = `
<DocumentFragment>
  <button
    data-testid="test-button-login"
    style="cursor: pointer;"
  >
    OK
  </button>
</DocumentFragment>
`;
